flow/core/kernel/toyota_energy.py

```diff
@@ -0,0 +1,71 @@
+import dill as pickle
+import boto3
+import botocore
+import math
+import random
+import statistics
+import numpy as np
+from scipy.interpolate import interp1d
+from collections import namedtuple
+from flow.energy.base_energy import BaseEnergyModel
+import os
+
+class ToyotaModel(BaseEnergyModel):
+
+    def __init__(self,kernel):
+        self.k = kernel
+
+        # download file from s3 bucket
+        s3 = boto3.client('s3')
+        s3.download_file('toyota.restricted',filename,'file.pkl') #move to init
+        with open('file.pkl','rb')as file:
+            self.toyota_energy = pickle.load(file) #self.prius_energy
+        # delete pickle file
+        os.remove(file.pkl)
+
+    def get_energy(self,veh_id,soc,grade):
+        pass
+
+
+class prius_energy(ToyotaModel):
+
+    super.__init__(kernel,filename = 'prius_test.pkl')

```

Comment:
sorry, i had this slightly wrong. this should be:
```
def __init__(self, kernel):
    super(PriusEnergy, self).__init__(kernel, filename = 'prius_test.pkl')
```
and I think all the parent/grandparent classes above should have the same call signature (i.e. `def __init__(self, kernel, filename=None):`